### DISCUSSION TITLE
How to stream autogen responses using FASTAPI ?

### DISCUSSION THREAD

**Discussion by AkashKulkarni4444** [microsoft/autogen#1537] [Q&A]
Does autogen have a streaming option/feature ?
Rather than just providing output all at once streaming should be present.
If it is present can it be integrated with fastAPI ?

**Comment by sonichi** [comment]
AutoGen Studio uses FastAPI.
cc @victordibia

**Comment by ekzhu** [comment]
We are working on a IOStream that supports streaming to any destination. #1551 @davorrunje

**Reply by davorrunje** [reply]
@AkashKulkarni4444 PR #1551 will be finished this week and it will have an example of streaming outputs using websockets integrated in FastAPI.

**Comment by victordibia** [comment]
Yes. [Autogen studio uses Fastapi](https://github.com/microsoft/autogen/blob/autogenstudio/samples/apps/autogen-studio/autogenstudio/web/app.py). 
And yes, the final implementation will explore the FastApi [websocket](https://fastapi.tiangolo.com/advanced/websockets/) native implementation.

**Reply by davorrunje** [reply]
@victordibia I'll ping you once I add FastAPI/websocket example to discuss if you need any extra support for Autogen Studio. I am working on it right now, should be soon.

**Reply by victordibia** [reply]
Thanks this will be great!

**Reply by sundar-sarvam** [reply]
Any updates on this? Can you share the reference please?

**Reply by sundar-sarvam** [reply]
I see it being non-streaming. Is there any way this will be streaming?
